var/spack/repos/builtin/packages/grass/package.py

```diff
@@ -194,13 +240,18 @@ def configure_args(self):
         else:
             args.append('--without-geos')
 
+        if '+x' in spec:
+            args.append('--with-x')
+        else:
+            args.append('--without-x')
+
         return args
 
     # see issue: https://github.com/spack/spack/issues/11325
     # 'Platform.make' is created after configure step
     # hence invoke the following function afterwards
     @run_after('configure')
     def fix_iconv_linking(self):
-        makefile = FileFilter('include/Make/Platform.make')
-        makefile.filter(r'^ICONVLIB\s*=\s*', 'ICONVLIB = -liconv')
-        return None
+        if self.spec.satisfies('@:7.6'):
+            makefile = FileFilter('include/Make/Platform.make')
+            makefile.filter(r'^ICONVLIB\s*=\s*', 'ICONVLIB = -liconv')
```

Comment:
This was causing linking errors like "no library found: -liconv-liconv", so I assume it was fixed in 7.8.0.

Reply:
The error you report above convinces me that 'iconv' issue is fixed in the new version. if you disable it for 7.8 and don't get any missing library 'iconv' error, then it should be fine.

Reply:
Actually, I just tried compiling 7.8 it seems in my case ICONVLIB is blank as opposed to being equla to '-iconv' as was also the case for version 7.6. So, for centos it seems to be necessary to do the filtering of Platform.make.